dotnet/samples/KernelSyntaxExamples/Example58_ConfigureRequestSettings.cs

```diff
@@ -0,0 +1,92 @@
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI;
+using Microsoft.SemanticKernel.SemanticFunctions;
+using RepoUtils;
+
+// ReSharper disable once InconsistentNaming
+public static class Example58_ConfigureRequestSettings
+{
+    /// <summary>
+    /// Show how to configure model request settings
+    /// </summary>
+    public static async Task RunAsync()
+    {
+        Console.WriteLine("======== Example57_ConfigureRequestSettings ========");
+
+        string serviceId = TestConfiguration.AzureOpenAI.ServiceId;
+        string apiKey = TestConfiguration.AzureOpenAI.ApiKey;
+        string chatDeploymentName = TestConfiguration.AzureOpenAI.ChatDeploymentName;
+        string endpoint = TestConfiguration.AzureOpenAI.Endpoint;
+
+        if (serviceId == null || apiKey == null || chatDeploymentName == null || endpoint == null)
+        {
+            Console.WriteLine("Azure serviceId, endpoint, apiKey, or deploymentName not found. Skipping example.");
+            return;
+        }
+
+        IKernel kernel = Kernel.Builder
+            .WithLoggerFactory(ConsoleLogger.LoggerFactory)
+            .WithAzureChatCompletionService(
+                deploymentName: chatDeploymentName,
+                endpoint: endpoint,
+                serviceId: serviceId,
+                apiKey: apiKey)
+            .Build();
+
+        var prompt = "Hello AI, what can you do for me?";
+
+        // Option 1:
+        // Invoke the semantic function and pass an OpenAI specific instance containing the request settings
+        var result = await kernel.InvokeSemanticFunctionAsync(prompt, requestSettings: new OpenAIRequestSettings() { MaxTokens = 256, Temperature = 0.7 });
+        Console.WriteLine(result.Result);
+
+        // Option 2:
+        // Load prompt template configuration including the request settings from a JSON payload
+        // Create the semantic functions using the prompt template and the configuration (loaded in the previous step)
+        // Invoke the semantic function using the implicitly set request settings
+        string configPayload = @"{
+          ""schema"": 1,
+          ""description"": ""Say hello to an AI"",
+          ""type"": ""completion"",
+          ""completion"": {
+            ""max_tokens"": 60,
+            ""temperature"": 0.5,
+            ""top_p"": 0.0,
+            ""presence_penalty"": 0.0,
+            ""frequency_penalty"": 0.0
+          }
+        }";
+        var templateConfig = JsonSerializer.Deserialize<PromptTemplateConfig>(configPayload);
+        var func = kernel.CreateSemanticFunction(prompt, config: templateConfig!, "HelloAI");
+
+        result = await kernel.RunAsync(func);
+        Console.WriteLine(result.Result);
+
+        /* OUTPUT (using gpt4):
+Hello! As an AI language model, I can help you with a variety of
+Hello! As an AI language model, I can help you with a variety of tasks, such as:
```

Comment:
Yes, the AI is called multiple times. First with a smaller max tokens and this is why the response is truncated.

Reply:
Got it. I was confused because it looks like there are two result outputs (line 46, line 68), but output in comments contains three different results (line 71, 72 and 84).

Reply:
nit: Also, looks like first call uses 256 tokens while second call uses 60, so the second output should be smaller in theory.

Reply:
Have fixed this